edb/pgsql/ast.py

```diff
@@ -367,6 +368,15 @@ class NullRelation(ReturningQuery):
     where_clause: BaseExpr
 
 
+@dataclasses.dataclass(frozen=True)
+class Param:
```

Comment:
Why isn't this a regular AST node?

Reply:
Because it's metadata and not part of SQL syntax, i.e. we don't codegen or otherwise handle it as a standalone node.

Reply:
What's the downside of still using an AST node for it?

Reply:
Mainly for dispatch purposes, we have `singledispatch` coverage tests and things like that.  I think it's cleaner to have non-AST structures to represent things that aren't encoding a piece of syntax.

Reply:
👍